public/app/features/dashboard/state/PanelModel.ts

```diff
@@ -162,6 +162,44 @@ export class PanelModel implements DataConfigSource {
       (this as any)[property] = model[property];
     }
 
+    const notPersistedPropertiesKeys = Object.keys(notPersistedProperties);
+    const mustKeepPropsKeys = Object.keys(mustKeepProps);
+    const defaultsKeys = Object.keys(defaults);
+    const properties = Object.keys(this);
+    for (const property of properties) {
+      if (notPersistedPropertiesKeys.find(prop => property === prop)) {
+        continue;
+      }
+
+      if (mustKeepPropsKeys.find(prop => property === prop)) {
+        continue;
+      }
+
+      if (defaultsKeys.find(prop => property === prop)) {
+        continue;
+      }
+
+      if (model[property]) {
+        continue;
+      }
+
+      if (!this.hasOwnProperty(property)) {
+        (this as any)[property] = undefined;
```

Comment:
think 
delete  (this as any)[property] 

is better but no sure :)

Reply:
Done!